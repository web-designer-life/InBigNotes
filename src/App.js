import React from 'react';
import './App.css';
import Container from './Container';

class App extends React.Component {
  render() {
      return (
          <div>
              <Container />
          </div>
      );
  }
}

export default App;
