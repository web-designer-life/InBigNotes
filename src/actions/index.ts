export const FETCH_NOTES_PENDING = 'FETCH_NOTES_PENDING';
export const FETCH_NOTES_SUCCESS = 'FETCH_NOTES_SUCCESS';
export const FETCH_NOTES_FAIL = 'FETCH_NOTES_FAIL';

export const ADD_NOTES_PENDING = 'ADD_NOTES_PENDING';
export const ADD_NOTES_SUCCESS = 'ADD_NOTES_SUCCESS';
export const ADD_NOTES_FAIL = 'ADD_NOTES_FAIL';

export const DELETE_NOTES_PENDING = 'DELETE_NOTES_PENDING';
export const DELETE_NOTES_SUCCESS = 'DELETE_NOTES_SUCCESS';
export const DELETE_NOTES_FAIL = 'DELETE_NOTES_FAIL';

export default { 
    FETCH_NOTES_PENDING,
    FETCH_NOTES_SUCCESS,
    FETCH_NOTES_FAIL,
    ADD_NOTES_PENDING, 
    ADD_NOTES_SUCCESS, 
    ADD_NOTES_FAIL, 
    DELETE_NOTES_PENDING, 
    DELETE_NOTES_SUCCESS, 
    DELETE_NOTES_FAIL,
};
