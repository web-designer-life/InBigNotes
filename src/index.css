body {
    font-family: 'Lora', 'Arial', sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    height: 100%;
    margin: 0;
    padding: 0;
}

code {
    font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New', monospace;
}

#root {
    display: flex;
    flex-direction: column;
    max-width: 1280px;
    height: 100%;
    margin: auto;
}
