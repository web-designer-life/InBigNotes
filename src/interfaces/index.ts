export interface INote {
    id: string;
    title: string;
    text: string;
    created_at: number;
    updated_at: number;
};
