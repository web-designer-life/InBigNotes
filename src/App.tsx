import * as React from 'react';
import { GlobalStyle } from './Components/Style/GlobalStyle';
import Container from './Container';
import './App.css';

class App extends React.Component {
    render() {
        return (
            <>
                <GlobalStyle />
                <Container />
            </>
        );
    }
};
  
export default App;
