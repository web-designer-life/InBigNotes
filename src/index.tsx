import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import history, { store } from './store/store';
import App from './containers/App';

ReactDOM.render(
    <React.StrictMode>
        <Provider store={store}>
            <App history={history} />
        </Provider>
    </React.StrictMode>,
    document.getElementById('root')
);
