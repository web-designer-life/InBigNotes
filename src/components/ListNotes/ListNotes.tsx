import React, { Component } from 'react';
import NoteComponent from '../NoteComponent';
import { List } from './style';
import { INote } from '../../interfaces';

interface Props {
    notes: INote[],
    selectedNotes: string[],
    addSelectedNote(note: INote): Function,
    removeUnselectedNote(note: INote): Function,
    navigateToPage(path: string): Function,
};

export default class ListNotes extends Component<Props> { 
    constructor(props) {
        super(props);
        
        this.state = {};
    }
    
    render() {
        const {
            notes, 
            selectedNotes, 
            addSelectedNote, 
            removeUnselectedNote, 
            navigateToPage,
        } = this.props;
        
        return (
            <List>
                {notes.map((note: INote) => (
                    <NoteComponent 
                        key={note.id}
                        note={note}
                        selectedNotes={selectedNotes}
                        addSelectedNote={addSelectedNote}
                        removeUnselectedNote={removeUnselectedNote}
                        navigateToPage={navigateToPage}
                    />
                ))}
            </List>
        );
    } 
}
