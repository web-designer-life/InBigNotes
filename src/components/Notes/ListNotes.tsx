import React, { Component } from 'react';
import NoteComponent from './NoteComponent';
import { List } from './style';
import { Note } from '../../interfaces';

interface Props {
    notes: Note[],
    selectedNotes?: string[],
    addSelectedNote(note: Note): Function,
    removeUnselectedNote(note: Note): Function,
    navigateToPage(path: string): Function,
};

export class ListNotes extends Component<Props> { 
    render() {
        const {
            notes, 
            selectedNotes, 
            addSelectedNote, 
            removeUnselectedNote, 
            navigateToPage,
        } = this.props;
        
        return (
            <List>
                {notes.map((note: Note) =>(
                    <NoteComponent 
                        key={note.id}
                        note={note}
                        selectedNotes={selectedNotes}
                        addSelectedNote={addSelectedNote}
                        removeUnselectedNote={removeUnselectedNote}
                        navigateToPage={navigateToPage}
                    />
                ))}
            </List>
        );
    } 
}
