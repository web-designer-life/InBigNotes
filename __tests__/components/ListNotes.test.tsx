import React from 'react';
import { shallow } from 'enzyme';
import ListNotes from '../../src/components/ListNotes';
import { INote } from '../../src/interfaces';

const ListNotesProps = {
    notes: [
        {
            id: 'b83cbac2-d74e-458d-85c0-75bdf9c29ba3',
            title: 'Army',
            text: 'An army, ground force, or land force is a fighting force that fights primarily on land.',
            created_at: 1632734670806,
            updated_at: 1632735522467,
        },
    ],
    selectedNotes: [
        'b83cbac2-d74e-458d-85c0-75bdf9c29ba3',
    ],
    addOrUpdateNote: (note: INote) => Function,
    addSelectedNote: (note: INote) => Function,
    removeUnselectedNote: (note: INote) => Function,
    navigateToPage: (path: string) => Function,
};

describe('ListNotes component', () => {
    it('should render ListNotes component', () => {
        const component = shallow(<ListNotes {...ListNotesProps} />);
        expect(component).toMatchSnapshot();
    });
});
