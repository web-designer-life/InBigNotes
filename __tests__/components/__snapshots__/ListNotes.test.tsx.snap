// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListNotes component should render ListNotes component 1`] = `
<styled.ul>
  <NoteComponent
    addSelectedNote={[Function]}
    key="b83cbac2-d74e-458d-85c0-75bdf9c29ba3"
    navigateToPage={[Function]}
    note={
      Object {
        "created_at": 1632734670806,
        "id": "b83cbac2-d74e-458d-85c0-75bdf9c29ba3",
        "text": "An army, ground force, or land force is a fighting force that fights primarily on land.",
        "title": "Army",
        "updated_at": 1632735522467,
      }
    }
    removeUnselectedNote={[Function]}
    selectedNotes={
      Array [
        "b83cbac2-d74e-458d-85c0-75bdf9c29ba3",
      ]
    }
  />
</styled.ul>
`;
